etc/scripts/timetable/old-stack-import/add-external-parts.js

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env node
+
+/**
+ * Copyright (c) 2014 "Fronteer LTD"
+ * Grasshopper Event Engine
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This script will read the external parts as outputted by the `django-export-external` command
+ * and merge them into the events tree that was generated by the `generate-tree` script
+ */
+
+var _ = require('lodash');
+var fs = require('fs');
+var yargs = require('yargs');
+
+var argv = yargs
+    .usage('Add parts who link to external timetables in the tree that holds the events.\nUsage: $0')
+    .example('$0 --events-tree events-tree.json --external-tree external-tree.json --output tree.json', 'Add parts with external data')
+
+    .demand('e')
+    .alias('e', 'events-tree')
+    .describe('e', 'The path where the events tree can be read')
+
+    .demand('x')
+    .alias('x', 'external-tree')
+    .describe('x', 'The path where the external tree can be read')
+
+    .demand('o')
+    .alias('o', 'output')
+    .describe('o', 'The path where the merged tree should be written')
+    .argv;
+
+// Read both trees
+var eventsTree = JSON.parse(fs.readFileSync(argv.e).toString('utf8'));
+var externalTree = JSON.parse(fs.readFileSync(argv.x).toString('utf8'));
+
+/**
+ * Find a node in a set of nodes
+ *
+ * @param  {Node[]}     nodes           The nodes to search through
+ * @param  {Node}       nodeToFind      The node to search for
+ * @return {Node}                       The matching node, or `null`
+ * @api private
+ */
+var _findNode = function(nodes, nodeToFind) {
+    var childNodes = _.values(nodes);
```

Comment:
I'm guessing `nodes` here isn't just a simple array of `Node`s. Can you fix up the type?